docs/src/workflow examples/Basic Neural Architecture Search/tuning.jl

```diff
@@ -0,0 +1,126 @@
+# # Neural Architecture Search with MLJFlux
+
+# Neural Architecture Search is (NAS) is an instance of hyperparameter tuning concerned with tuning model hyperparameters
+# defining the architecture itself. Although it's typically performed with sophisticated search algorithms for efficiency,
+# in this example we will be using a simple random search.
+
+using Pkg     #src
+Pkg.activate(@__DIR__);     #src
+Pkg.instantiate();     #src
+
+# **Julia version** is assumed to be 1.10.*
+
+# ### Basic Imports
+
+using MLJ               # Has MLJFlux models
+using Flux              # For more flexibility
+using RDatasets: RDatasets        # Dataset source
+using DataFrames        # To view tuning results in a table
+
+# ### Loading and Splitting the Data
+
+iris = RDatasets.dataset("datasets", "iris");
+y, X = unpack(iris, ==(:Species), colname -> true, rng = 123);
+X = Float32.(X);      # To be compatible with type of network network parameters
+first(X, 5)
+
+
+
+# ### Instantiating the model
+
+# Now let's construct our model. This follows a similar setup the one followed in the [Quick Start](../../index.md).
+NeuralNetworkClassifier = @load NeuralNetworkClassifier pkg = "MLJFlux"
+clf = NeuralNetworkClassifier(
+	builder = MLJFlux.MLP(; hidden = (1, 1, 1), σ = Flux.relu),
+	optimiser = Flux.ADAM(0.01),
+	batch_size = 8,
+	epochs = 10,
+	rng = 42,
+)
+
+
+# ### Generating Network Architectures
+# We know that the MLP builder takes a tuple of the form $(z_1, z_2, ..., z_k)$ to define a network with $k$ hidden layers and
+# where the ith layer has $z_i$ neurons. We will proceed by defining a function that can generate all possible networks with a
+# specific number of hidden layers, a minimum and maximum number of neurons per layer and increments to consider for the number of neurons.
+
+function generate_networks(;
+	min_neurons::Int,
+	max_neurons::Int,
+	neuron_step::Int,
+	num_layers::Int,
+)
+    ## Define the range of neurons
+	neuron_range = min_neurons:neuron_step:max_neurons     
+
+    ## Empty list to store the network configurations
+	networks = Vector{Tuple{Vararg{Int, num_layers}}}()    
+
+	## Recursive helper function to generate all combinations of tuples
+	function generate_tuple(current_layers, remaining_layers)
+		if remaining_layers > 0
+			for n in neuron_range
```

Comment:
nit picking: I think a indentation of 4 spaces is the Blue Style standard. Do not use TABs as Literate and Documenter don't seem to render them as expected.

Reply:
Will try to see if I can tell my Julia formatter to use spaces. I noticed that it incorrectly renders comments which is why I made the spacing non uniform for the comments.